### PR TITLE
[SRE-90] Convert to haskell.nix

### PR DESCRIPTION
**Pull request by craigem**

### CODE REVIEW

**Comment by dnadales** [comment]
@craigem let me know when this is ready for us to approve and merge :pray:

File: shell.nix [hunk 1]
```diff
@@ -1 +1,65 @@
-(import ./default.nix {}).shell
+# This file is used by nix-shell.
+# It just takes the shell attribute from default.nix.
```
Comment (sevanspowell) [inline shell.nix:2]: Super super minor, I don't think this comment applies here?

File: release.nix [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
+#
+# Hydra release jobset.
+#
+# The purpose of this file is to select jobs defined in default.nix and map
+# them to all supported build platforms.
+#
+############################################################################
+
+# The project sources
+{ decentralized-updates ? { outPath = ./.; rev = "abcdef"; }
```
Comment (sevanspowell) [inline release.nix:11]: What does choosing this "rev" do?

File: nix/haskell.nix [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        # split data output for ekg to reduce closure size
+        packages.ekg.components.library.enableSeparateDataOutput = true;
+        packages.cardano-node.configureFlags = [ "--ghc-option=-Werror" ];
+        packages.cardano-config.configureFlags = [ "--ghc-option=-Werror" ];
+        enableLibraryProfiling = profiling;
+      }
+      (lib.optionalAttrs stdenv.hostPlatform.isWindows {
+        # Disable cabal-doctest tests by turning off custom setups
+        packages.comonad.package.buildType = lib.mkForce "Simple";
```
Comment (sevanspowell) [inline nix/haskell.nix:56]: Are all these overrides required by the project? and if not, do we typically include them all anyway?

**Comment by sevanspowell** [comment]
Reviewed as much as I can without looking at one of these before, I've asked a bunch of questions because I haven't seen any haskell.nix projects before.

File: shell.nix (hunk 1 above)
Comment (craigem) [reply shell.nix:2]: Removed line 2 from pending commit.

**Comment by johnalotoski** [comment]
This PR is creating 1100+ dead pids on hydra for an extended period of time.  Disabling hydra eval/building temporarily until further work on the PR is resumed and this can be investigated.  Ex:
```
# pgrep -f decentralized | xargs -I {} bash -c 'pstree -p {} | wc -l'
5
1267
1267

# ps aux | grep [1]19507
hydra    119507  1.0  0.0      0     0 ?        Z    21:33   0:04 [hydra-eval-jobs] <defunct>
```

**Comment by dnadales** [comment]
This seems to be abandoned. I did the nix migration myself. CC: @iohk-devops

**Pull request closed without merging**